### ISSUE TITLE
[Question] Why logging files (monitor.csv, progress.csv) also consider the evaluation record? 

### ISSUE THREAD

**Issue by olenmg** [DLR-RM/stable-baselines3#852]
### Question

Hello, thanks for your great contribution! SB3 is a perfect starting point for me to do experiments with an RL environment. 

Now I am trying to analyze the log files without the help of Zoo, and I arise with a question about `Monitor` wrapper, especially for a logging file, monitor.csv.
From what I've seen, monitor.csv not only includes the training log but that of evaluation.
i.e., monitor.csv must write the log when "episode-done" occurred regardless of whether that episode has proceeded as train or eval.

So I'm wondering why `Monitor` wrapper put the info about evaluation between the train history. 
I believe that evaluation logs can disturb the analysis or plotting of training progress if it is logged in the current form.

Is it because there is no difference in the environment between train and eval?
However, I think that the multiple lines of the evaluation log still can badly affect the analysis of data since there is no policy update on that step (and in another step, do an update).
And the same question goes for progress.csv with SB3 `Logger` (There's an evaluation log in between here, too.)

So, in short, is it reasonable to log & analyze the info about train and eval at the same time?

Thanks!

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by araffin** [comment]
>  From what I've seen, monitor.csv not only includes the training log but that of evaluation.

What makes you think that?
`x.monitor.csv` only contains info about training in the zoo. The evaluation stats are in `evaluations.npz` and are generated using the `EvalCallback`.

> Is it because there is no difference in the environment between train and eval?

as written in the doc, you should use different env between  train and eval, otherwise the monitor file contain both infos, as the monitor wrapper is attached to one instance of the env.

**Comment by olenmg** [comment]
Oh, I see.. thanks for the fast reply.

I was confused because in the [tutorial](https://colab.research.google.com/github/araffin/rl-tutorial-jnrr19/blob/sb3/1_getting_started.ipynb) they used the same environment for train & eval :joy: 

Then, is the following code valid? 

```python
env = make_vec_env("CartPole-v1", ...)
eval_env = make_vec_env("CartPole-v1", ...)
```

Otherwise, progress.csv (of SB3 Logger) still has the same issue. Doesn't it matter?


> Because most algorithms use exploration noise during training, you need a separate test environment to evaluate the performance of your agent at a given time.

And.. You may talk about the above. In fact, apart from the logging issue, I didn't fully understand what the document said. Doesn't the environment work as same even if we separate eval and train? I cannot understand what 'exploration noise' exactly means. Isn't it dependent on the agent's exploration, not on the environment? Please give some advice on this point...

**Comment by araffin** [comment]
> Then, is the following code valid?

yes, if you use the RL Zoo, this is done automatically for you.

> Otherwise, progress.csv (of SB3 Logger) still has the same issue. Doesn't it matter?

Actually, I'm not sure what `progress.csv` records...

> I was confused because in the [tutorial](https://colab.research.google.com/github/araffin/rl-tutorial-jnrr19/blob/sb3/1_getting_started.ipynb) they used the same environment for train & eval joy

yes... should be updated. I think I did that to make it simple.

> I cannot understand what 'exploration noise' exactly means.

You can take a look at spinning up tutorial for that or also [this paper](https://proceedings.mlr.press/v164/raffin22a.html)

> Isn't it dependent on the agent's exploration, not on the environment? 

at test time, unless you need some stochasticity, the exploration should be removed, this may have a great impact on performance.

>  Doesn't the environment work as same even if we separate eval and train? 

yes, but if you evaluate with the same env in the middle of training, it will change the env state and may have undesirable side effects.
